packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,473 @@
+     */
+    TRIANGLES_VERTICES = 1,
+    /**
+     * A checkerboard grid of the mesh's UV set 0
+     */
+    UV0 = 2,
+    /**
+     * A checkerboard grid of the mesh's UV set 1
+     */
+    UV1 = 3,
+    /**
+     * The mesh's vertex colors displayed as the primary texture
+     */
+    VERTEXCOLORS = 4,
+    /**
+     * An arbitrary, distinguishable color to identify the material
+     */
+    MATERIALIDS = 5,
+    /**
+     * Points drawn over vertices of mesh
+     * NOTE: For this mode to work correctly, MeshDebugPluginMaterial.prepareMeshForTrianglesAndVerticesMode() must first be called on mesh.
+     */
+    VERTICES = 6,
+}
+
+/**
+ * Options for MeshDebugPluginMaterial that are given at initialization
+ */
+export interface MeshDebugOptions {
+    /**
+     * Mesh debug mode to initialize with.
+     * Defaults to NONE.
+     */
+    mode?: MeshDebugMode;
+    /**
+     * Whether the mesh debug visualization should multiply with any PBR debug colors underneath.
+     * Defaults to true.
+     */
+    multiplyDebug?: boolean;
+    /**
+     * Default diffuse color of the visualization.
+     * Defaults to (1.0, 1.0, 1.0).
+     */
+    shadedDiffuseColor?: Vector3;
+    /**
+     * Default specular color of the visualization.
+     * Defaults to (0.8, 0.8, 0.8).
+     */
+    shadedSpecularColor?: Vector3;
+    /**
+     * How dark the shading of the visualization is.
+     * Defaults to 10.
+     */
+    shadedSpecularPower?: number;
+    /**
+     * Width of edge lines in TRIANGLES and TRIANGLE_VERTICES modes.
+     * Defaults to 0.7.
+     */
+    wireframeThickness?: number;
+    /**
+     * Color of edge lines in TRIANGLES mode, and color of dots in TRIANGLES_VERTICES and VERTICES mode.
+     * Defaults to (0.0, 0.0, 0.0).
+     */
+    wireframeColor1?: Vector3;
+    /**
+     * Color of edge lines in TRIANGLES_VERTICES modes.
+     * Defaults to (0.8, 0.8, 0.8).
+     */
+    wireframeColor2?: Vector3;
+    /**
+     * Radius of dots drawn over vertices in TRIANGLE_VERTICES and VERTICES mode.
+     * Defaults to 1.2.
+     */
+    vertexRadius?: number;
+    /**
+     * Size of tiles in UV1 or UV2 modes.
+     * Defaults to 20.
+     */
+    uvScale?: number;
+    /**
+     * 1st color of checkerboard grid in UV1 or UV2 modes.
+     * Defaults to (1.0, 1.0, 1.0).
+     */
+    uvColor1?: Vector3;
+    /**
+     * 2nd color of checkerboard grid in UV1 or UV2 modes.
+     * Defaults to (0.5, 0.5, 0.5).
+     */
+    uvColor2?: Vector3;
+    /**
+     * Identifying color of this material in MATERIALIDS mode.
+     * Defaults to a randomly-generated color.
+     */
+    materialColor?: Vector3;
+    /**
+     * Multiplicand for the square root of the camera distance in TRIANGLES_VERTICES and VERTICES mode.
+     * Defaults to .001
+     */
+    distanceScale?: number;
+}
+
+/** @internal */
+export class MeshDebugDefines extends MaterialDefines {
+    dbg_MODE: MeshDebugMode = MeshDebugMode.NONE;

```

Comment:
I would use `DBG` instead of `dbg`, as the convention for a define name is that it is all in upper case.